### ISSUE TITLE
[Feature] Allow injecting logging metadata (MDC), e.g. when jobs throw exceptions

### ISSUE THREAD

**Issue by patkujawa-wf** [jobrunr/jobrunr#326]
**Is your feature request related to a problem? Please describe.**
[This code](https://github.com/jobrunr/jobrunr/blob/37e41ff59c78f147bed66266e5d22a9313d10375/core/src/main/java/org/jobrunr/server/BackgroundJobPerformer.java#L56) runs when a job throws an exception, but I cannot find a way to inject [Mapped Diagnostic Context (MDC)](https://www.baeldung.com/mdc-in-log4j-2-logback) so that that log message contains some important metadata.

**Describe the solution you'd like**
I would like a hook/API that allows intercepting or decorating logs that jobrunr emits so I can wrap them in MDC like `{"username": "Pat"}`.

Maybe some way to tie into `ThreadLocalJobContext` / `JobContext` with a specific logging/MDC context?

**Describe alternatives you've considered**
1. Catching and rethrowing exceptions after I've logged them myself. Results in duplicate logs and jobrunr logs are still hard to correlate. (Or I suppose we could turn off jobrunr logs, but I'm hesitant to do that.)
2. Include metadata in the job id - can't, has to be a UUID.
3. Include metadata in the job name - is that even possible?
4. Add a logging filter and try to pull the info from the job metadata?
5. Use a custom `JobActivator`? (Not sure how that'd work along with Spring DI.)

**Additional context**
Example log in splunk for our system (which only has the metadata injected by logback config, not the per-request metadata I'd like to see):

```
   level: warning
   message: Job(id=53252c34-79d2-4232-92c1-346d6fbc42c8, jobName='<removed>.FooRequestHandler.run(<removed>FooJobRequest@786b27e3)') processing failed: An exception occurred during the performance of the job
   metadata: { [-]
     library_name: <removed>
     lineNumber: org.jobrunr.server.BackgroundJobPerformer.updateJobStateToFailedAndRunJobFilters(BackgroundJobPerformer.java:115)
     threadName: backgroundjob-worker-pool-11-thread-5
   }
   name: org.jobrunr.server.BackgroundJobPerformer
```

**Comment by rdehuyss** [comment]
Feel free to open a PR for this.

**Comment by patkujawa-wf** [comment]
Thanks, I'll have to think through what that API would look like. Am I correct that there's no simple way to do it today? I wasn't sure if my assumptions were correct :)

**Comment by rdehuyss** [comment]
No, I don't think there is a simple way to do. The more I think about it, the more is see added complexity as the Job is launched in a different thread with no access to the the MDC from where the job was launched.

**Comment by rdehuyss** [comment]
If this feature would be added to the Pro version, would it justify purchasing a license?

**Comment by patkujawa-wf** [comment]
@rdehuyss 😆 I like how you think! But no, it wouldn't be the killer feature that prompts purchasing a license, it's more of a nice-to-have, sorry.

I was thinking that the MDC could be included with the job as part of its metadata and then injected in the thread while the job runs (and popped out when the job fails/finishes).

**Comment by rdehuyss** [comment]
Solved in v5.0.0 branch.
